src/preProcessPattern.js

```diff
@@ -30,6 +31,18 @@ export default function preProcessPattern(globalRef, pattern) {
         }
       : Object.assign({}, pattern);
 
+  if (typeof pattern.from === 'string') {
+    pattern.from = normalizePath(pattern.from);
+  } else if (
+    typeof pattern.from === 'object' &&
+    typeof pattern.from.glob === 'string'
+  ) {
+    pattern.from = {
+      ...pattern.from,
+      glob: normalizePath(pattern.from.glob),
+    };
+  }
```

Comment:
Find one interesting thing https://github.com/floatdrop/gulp-watch/blob/master/index.js#L54 , we need add test for '!'

Reply:
Maybe best place for this here https://github.com/webpack-contrib/copy-webpack-plugin/blob/master/src/utils/escape.js?

Reply:
Sorry, I didn't get it.

Do you mean test `'[!]\\hello.txt'`?

Reply:
Something like this `!(directory)/**/*.txt`